htdocs/core/actions_extrafields.inc.php

```diff
@@ -127,6 +128,28 @@
     			}
     		}
         }
+		if (! $error)
+		{
+			$cats = array ();
+			// attrname must not be a reserved word
+			if ($db->type == 'pgsql')
+				$sql ="SELECT word as reservedword FROM pg_get_keywords() WHERE catcode='R'";
+			else
+				$sql ="SELECT name as reservedword FROM mysql.help_keyword";
```

Comment:
This are system table.
And the database user, if server setup was done correctly won't have access to such tables.
A hardcoded list of common keys should be used even if it does not prevent 100% of cases.